### PR TITLE
README for Vault auth path configuration 

### PR DESCRIPTION
**Pull request by revit13**
This PR adds a README file with a description of Vault auth path configuration. It is added temporally until an appropriate documentation will be created.

### CODE REVIEW

File: pkg/vault/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+**Prerequisite for multi-cluster setup:**
+
+1. Vault should have connectivity to the remote clusters to allow it to communicate to the clusters API server.
+2. An [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resource should be created in the coordinator cluster to enable communication to Vault service from the remote clusters.
+
+
+Some of the steps described below are not specific to the Mesh for Data project but rather are Vault specific and can be found in Vault related online tutorials. The steps were tested on Kind cluster version 8.28 and OpenShift Container Platform 4.5.
+
+It is assumed that Vault is deployed in the coordinator cluster in `m4d-system` namespace prior to executing the steps.
```
Comment (ronenkat) [inline pkg/vault/README.md:17]: It is assumed that --> the following steps assumes that 
prior to executing the steps. --->  <delete>

File: pkg/vault/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Module authentication is done by configuring Vault to use [Kubernetes auth method](https://www.vaultproject.io/docs/auth/kubernetes) in each cluster. Using this method the modules can authenticate to Vault by providing their service account token. Behind the scenes Vault authenticates the token by submitting TokenReview request to the API server of the kubernetes cluster where the module is running. 
+
+**Prerequisite for multi-cluster setup:**
+
+1. Vault should have connectivity to the remote clusters to allow it to communicate to the clusters API server.
+2. An [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resource should be created in the coordinator cluster to enable communication to Vault service from the remote clusters.
+
+
+Some of the steps described below are not specific to the Mesh for Data project but rather are Vault specific and can be found in Vault related online tutorials. The steps were tested on Kind cluster version 8.28 and OpenShift Container Platform 4.5.
```
Comment (ronenkat) [inline pkg/vault/README.md:15]: Move 
> The steps were tested on Kind cluster version 8.28 and OpenShift Container Platform 4.5.

to the top of the readme.

File: pkg/vault/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+It is assumed that Vault is deployed in the coordinator cluster in `m4d-system` namespace prior to executing the steps.
+
+1. Setup [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) to Vault service in the coordinator cluster for multi-cluster setup:
```
Comment (ronenkat) [inline pkg/vault/README.md:19]: is this a duplicate of (2) above?

File: pkg/vault/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+The ingress should be deployed in the `m4d-system` namespace where Vault is deployed.
+
+2. Add Vault address to `m4d-config`:
+The address of Vault should be added to the kubernetes ConfigMap called `m4d-config` in `m4d-system` namespace in the coordinator cluster as follows:
+```bash
+VAULT_ADDRESS: <Vault address>
+```
+In a multi-cluster setup the Vault address should be the Ingress address described above while in single cluster it should set to: `http://vault.m4d-system:8200/`.
+
+**Pre-steps for configuring Vault:**
```
Comment (ronenkat) [inline pkg/vault/README.md:30]: maybe change as follows? Pre-steps for configuring Vault: --> Enabling Kubernetes auth for each cluster with running M4D modules ?

File: pkg/vault/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+The next steps describe how to configure Vault to use Kubernetes auth method for each cluster. To do that, Vault root token should be used in order to execute the configuration commands against Vault. The root token can be extracted by the following commands executed in the coordinator cluster:
+```bash
```
Comment (ronenkat) [inline pkg/vault/README.md:32]: can you make it as steps? (i), (ii) instead of having text "after that".... ?

**Comment by revit13** [comment]
@ronenkat Thanks for the review comments. I made changes based on them.

**Comment by roee88** [comment]
@revit13 we still need the README right? Do you want to rebase and merge? Sorry I missed the progress on this one

**Comment by roee88** [comment]
@revit13 please clarify what is the status of this PR.

**Comment by roee88** [comment]
@revit13 please advise

**Comment by roee88** [comment]
@revit13 reopen if needed

**Pull request closed without merging**